drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -519,19 +520,25 @@ static int nrf5_tx(const struct device *dev,
 		/* Handle ACK packet. */
 		return handle_ack(nrf5_radio);
 	case NRF_802154_TX_ERROR_NO_MEM:
-		return -ENOBUFS;
+		result = -ENOBUFS;
 	case NRF_802154_TX_ERROR_BUSY_CHANNEL:
-		return -EBUSY;
+		result = -EBUSY;
 	case NRF_802154_TX_ERROR_INVALID_ACK:
 	case NRF_802154_TX_ERROR_NO_ACK:
-		return -ENOMSG;
+		result = -ENOMSG;
 	case NRF_802154_TX_ERROR_ABORTED:
 	case NRF_802154_TX_ERROR_TIMESLOT_DENIED:
 	case NRF_802154_TX_ERROR_TIMESLOT_ENDED:
-		return -EIO;
+	default:
+		result = -EIO;
 	}
 
-	return -EIO;
+#if NRF_802154_ENCRYPTION_ENABLED
+	/* Copy secured payload back to the upper layer, if TX was not successful. */
```

Comment:
It would be actually nice to know why do we need to copy the payload back (both here and in the commit message). Now you are just saying what the memcpy does but there is no reason why do we need to do this in the first place.

Reply:
I have added comments explaining why I added this.